src/types/homepage.ts

```diff
@@ -20,6 +20,16 @@ export type HighlightOption = {
   url: string
 }
 
+export type AnnouncementOption = {
+  title: string
+  date: string
+  announcement: string
+  link_text: string
+  link_url: string
+}
+
+export type AnnouncementError = AnnouncementOption
```

Comment:
not important _now_ but hte lack of a discriminant property (eg: `kind: 'error' | 'option'`) means that checking if this is actualy error or not is upheld by devs rather than the type system.

Reply:
ok, to prior if we touch this part in the future + find it painful ya